core/src/main/scala/akka/persistence/r2dbc/state/scaladsl/DurableStateDao.scala

```diff
@@ -328,13 +536,25 @@ private[r2dbc] class DurableStateDao(settings: R2dbcSettings, connectionFactory:
                 stmt
             }
           }
+
+          r2dbcExecutor.withConnection(s"delete [$persistenceId]") { connection =>
+            for {
+              updatedRows <- R2dbcExecutor.updateOneInTx(updateStatement(connection))
+              _ <- changeHandlers.get(entityType) match {
+                case None => FutureDone
+                case Some(handler) =>
+                  val session = new R2dbcSession(connection)
+                  handler.process(session, change)
```

Comment:
Another place where adding details to exceptions could be useful, and another one for hard deletes below.